### DISCUSSION TITLE
Empty vehicle after last job

### DISCUSSION THREAD

**Discussion by kaiSchweegmann** [reinterpretcat/vrp#117] [Q&A] [answered]
Hi,

I am currently trying to solve a "garbage collection"-like problem.
My vehicle drives to different jobs and collects trash.
When the capacity of the vehicle is reached it stops at a pre-defined depots.
At the moment the vehicle drives to the end location after collecting at each job with no regard to the current amount of trash loaded on the vehicle.

Is it possible to edit the problem, so that the vehicle unloads at a depot after the last job is completed, in order to return to the end position completely empty?

My Problem: `{
	"plan": {
		"jobs": [
			{
				"id": "1",
				"pickups": [
					{
						"demand": [
							120
						],
						"places": [
							{
								"duration": 0,
								"location": {
									"lat": 52.00627,
									"lng": 8.56765
								}
							}
						]
					}
				]
			},
			{
				"id": "2",
				"pickups": [
					{
						"demand": [
							120
						],
						"places": [
							{
								"duration": 0,
								"location": {
									"lat": 52.01441,
									"lng": 8.54920
								}
							}
						]
					}
				]
			}
		]
	},
	"fleet": {
		"profiles": [
			{
				"name": "car"
			}
		],
		"vehicles": [
			{
				"costs": {
					"time": 0.01,
					"fixed": 25,
					"distance": 0.0001
				},
				"shifts": [
					{
						"end": {
							"latest": "2023-08-10T07:00:00Z",
							"location": {
								"lat": 52.01697,
								"lng": 8.55454
							}
						},
						"start": {
							"latest": "2023-07-10T07:00:00Z",
							"earliest": "2023-07-10T07:00:00Z",
							"location": {
								"lat": 52.01697,
								"lng": 8.55454
							}
						},
						"breaks": [],
						"reloads": [
							{
								"duration": 0,
								"location": {
									"lat": 52.01757,
									"lng": 8.56358
								}
							}
						]
					}
				],
				"typeId": "778",
				"profile": {
					"scale": 1,
					"matrix": "car"
				},
				"capacity": [
					140
				],
				"vehicleIds": [
					"778"
				]
			}
		]
	},
	"objectives": [
		[
			{
				"type": "minimize-unassigned"
			}
		],
		[
			{
				"type": "minimize-arrival-time"
			}
		],
		[
			{
				"type": "minimize-tours"
			}
		],
		[
			{
				"type": "balance-duration",
				"options": {
					"threshold": 0.05
				}
			}
		],
		[
			{
				"type": "minimize-cost"
			}
		]
	]
}`

**Comment by reinterpretcat** [comment, marked as answer]
Sorry for late answer, just noticed it.

In case of closed VRP, the last location is considered the point where all goods are unloaded and no further work is needed. I would recommend you to apply some post-processing: given that depot and end locations are known, you can calculate duration between them and deduct this from shift time. Later, in your software, add end location to the tour (and statistic)
